### PR TITLE
allow sizes prop to accept an array

### PR DESCRIPTION
**Pull request by gbellini911**
https://5a85e56213417800200978ab-zetreramsm.chromatic.com/?path=/story/stamp--pass-an-array-of-sizes

Update `Stamp` so its size property can also accept an array 

_(Shoutout: @pcln-james helped out immensely with this)_

### CODE REVIEW

File: packages/core/src/Stamp/Stamp.stories.js [hunk 1]
```diff
@@ -87,3 +87,67 @@ storiesOf('Stamp', module)
...
+    <div>
+      <Stamp
+        borderColor='blue'
+        bg='blue'
+        color='white'
+        mr={2}
+        size={['medium', null, null, null, null, null]}
+      >
+        <PinIcon />
+        <Text>Medium at xs</Text>
```
Comment (craigpalermo) [inline packages/core/src/Stamp/Stamp.stories.js:100]: I checked the updated storybook in Chromatic, which works great, but can we change these to `Large at xyz`? That'd make it a bit easier to see the differences.

File: packages/core/src/utils/utils.js [hunk 2]
```diff
@@ -108,23 +109,49 @@ export const getContrastRatio = (colorA, colorB) => {
   return (Math.max(luminA, luminB) + 0.05) / (Math.min(luminA, luminB) + 0.05)
 }
 
+export const getBreakpointSize = (array, length) => {
+  const filteredArray = array.slice(0, length)
+  return filteredArray.reduceRight((carry, item) => {
+    return !carry ? item : carry
+  }, null)
```
Comment (craigpalermo) [inline packages/core/src/utils/utils.js:116]: Can we add some comments describing the purpose of each of the steps in this function? It'll make it easier to maintain.

**Review by craigpalermo — CHANGES_REQUESTED** [review]
Looking great so far! Can you also run `rush change`?

File: packages/core/src/utils/utils.js (hunk 2 above)
Comment (gbellini911) [reply packages/core/src/utils/utils.js:126]: Updated 👍 (open to adding even more detail if you think necessary)

**Review by craigpalermo — APPROVED** [review]
LGTM, nice work!

File: packages/core/src/Stamp/Stamp.js [hunk 3]
```diff
@@ -66,8 +66,8 @@ const Stamp = styled.div`
   color: ${getPaletteColor('base')};
 
   ${space} ${fontSize};
-  ${applySizes(sizes)}
-  ${applyVariations('Stamp', variations)}
+  ${applySizes(sizes)};
+  ${applyVariations('Stamp', variations)};
 `
 
 Stamp.displayName = 'Stamp'
```
Comment (craigpalermo) [inline packages/core/src/Stamp/Stamp.js:73]: JW, do we need to make any changes to the `sizes` prop type, or add any additional values?

**Comment by craigpalermo** [comment]
I think we need to verify that the default behavior still works when no size prop is provided. This storybook diff appears to have changed for the default.

https://www.chromatic.com/snapshot?appId=5a85e56213417800200978ab&id=5fa41c93e8b10200221cb26f

**Comment by gbellini911** [comment]
> I think we need to verify that the default behavior still works when no size prop is provided. This storybook diff appears to have changed for the default.
> 
> https://www.chromatic.com/snapshot?appId=5a85e56213417800200978ab&id=5fa41c93e8b10200221cb26f

@craigpalermo split the 'breaking' line height change into a separate PR

File: packages/core/src/utils/utils.spec.js [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+  getLuminance,
   getPaletteColor,
-  hasPaletteColor,
   getTextColorOn,
-  getByPalette,
-  color,
-  borders,
-  createTheme,
+  hasPaletteColor,
+  hexToRgb,
```
Comment (bmansell) [inline packages/core/src/utils/utils.spec.js:17]: Import alphabetizing 🙌

File: packages/core/src/Stamp/Stamp.stories.js [hunk 5]
```diff
@@ -87,3 +87,67 @@ storiesOf('Stamp', module)
       </Stamp>
     </div>
   ))
+  .add('Pass an array of sizes', () => (
+    <div>
+      <Stamp
+        borderColor='blue'
+        bg='blue'
+        color='white'
```
Comment (bmansell) [inline packages/core/src/Stamp/Stamp.stories.js:95]: Could we switch these to use palette colours instead?
```suggestion
        color='primary'
```

**Review by bmansell — CHANGES_REQUESTED** [review]

File: packages/core/src/Stamp/Stamp.stories.js (hunk 5 above)
Comment (craigpalermo) [reply packages/core/src/Stamp/Stamp.stories.js:95]: I'm not sure exactly how it's supposed to work, but this is what `color='primary'` looks like on Stamp (far left):
Screen Shot 2020-11-06 at 3 55 09 PM

Comment (craigpalermo) [reply packages/core/src/Stamp/Stamp.stories.js:95]: @pcln-brianna , just saw your update. That looks as expected 👍

Comment (bmansell) [reply packages/core/src/Stamp/Stamp.stories.js:95]: Yeah I realized it was the reverse of what I was thinking after checking our docs haha. Tested the updated suggestion:
Screen Shot 2020-11-06 at 2 57 35 PM

**Review by bmansell — APPROVED** [review]

**Pull request merged**